### PR TITLE
Users/maburson/yarn and workspaces

### PR DESCRIPTION
**Pull request by maxburs**
# Changes
## Replace NPM with Yarn
Every time I've run `npm i` in this repo I've gotten a different lockfile than was checked in. Devs working in this repo are using Yarn in a private repo with good results.
## Add /example/react and /package to Yarn workspace
Yarn systemlinks packages together if they're in the same workspace, and shares node_modules when possible. This will reduce `yarn install` times, and allow us to easily run examples with the local /package version.

A follow up PR will use this to move the test app out of /package and make it it's own project.

### CODE REVIEW

File: package/tsconfig.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+        "outDir": "./out/amd",
+        "target": "ES2020",
+        "alwaysStrict": true,
+        "declaration": true,
+        "skipLibCheck": true,
+        "types": [
+            "monaco-editor-core/monaco",
+            "@kusto/language-service/Kusto.JavaScript.Client",
+            "@kusto/language-service-next/Kusto.Language.Bridge"
+        ]
```
Comment (maxburs) [inline package/tsconfig.json:17]: Removed triple slash directives in favor of this, as per the TypeScript docs recommendation: https://www.typescriptlang.org/docs/handbook/triple-slash-directives.html#-reference-types-

File: package/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
+{
+    "$schema": "https://json.schemastore.org/tsconfig",
+    "include": ["./src/**/*"],
+    "compilerOptions": {
+        "rootDir": "./src",
+        "module": "amd",
+        "moduleResolution": "node",
+        "outDir": "./out/amd",
+        "target": "ES2020",
```
Comment (maxburs) [inline package/tsconfig.json:9]: Was es5. I'm assuming updating this to something a little newer is fine. It should make debugging a fair bit easier.

Comment (maxburs) [reply package/tsconfig.json:9]: r.js Failed while parsing because of object spread (`...`). I've updated it to the latest version it successfully parses.

**Review by noam-shapira — CHANGES_REQUESTED** [review]
add .yarn to git ignore

File: package/tsconfig.json (hunk 2 above)
Comment (noam-shapira) [reply package/tsconfig.json]: Need to keep the es5, it might break consumers (Synapse for sure, Ibiza portal, and more)

Comment (maxburs) [reply package/tsconfig.json]: Bummer. I've reverted the target to es5

**Comment by maxburs** [comment]
> add .yarn to git ignore

I've gitignored the yarn cache as discussed on Teams

**Review by noam-shapira — APPROVED** [review]

**Pull request merged**